src/storage/hummock_sdk/src/compaction_group/mod.rs

```diff
@@ -46,11 +46,215 @@ impl From<StaticCompactionGroupId> for CompactionGroupId {
 
 pub mod group_split {
     use std::cmp::Ordering;
+    use std::collections::BTreeSet;
+
+    use bytes::Bytes;
+    use risingwave_common::catalog::TableId;
+    use risingwave_common::hash::VirtualNode;
+    use risingwave_pb::hummock::PbLevelType;
 
     use super::hummock_version_ext::insert_new_sub_level;
-    use crate::can_concat;
+    use super::StateTableId;
+    use crate::key::{FullKey, TableKey};
+    use crate::key_range::KeyRange;
     use crate::level::{Level, Levels};
+    use crate::sstable_info::SstableInfo;
+    use crate::{can_concat, HummockEpoch};
+
+    /// The split will follow the following rules:
+    /// 1. Ssts with `split_key` will be split into two separate ssts and their `key_range` will be changed `sst_1`: [`sst.key_range.right`, `split_key`) `sst_2`: [`split_key`, `sst.key_range.right`].
+    /// 2. Currently only `vnode` 0 and `vnode` max is supported.
+    /// 3. Due to the above rule, `vnode` max will be rewritten as `table_id` + 1, vnode 0
+
+    // By default, the split key is constructed with vnode = 0 and epoch = 0, so that we can split table_id to the right group
+    pub fn build_split_key(table_id: StateTableId, vnode: VirtualNode) -> Bytes {
+        build_split_full_key(table_id, vnode).encode().into()
+    }
+
+    /// generate a full key for convenience to get the `table_id` and `vnode`
+    pub fn build_split_full_key(
+        mut table_id: StateTableId,
+        mut vnode: VirtualNode,
+    ) -> FullKey<Vec<u8>> {
+        if VirtualNode::MAX == vnode {
+            // Modify `table_id` to `next_table_id` to satisfy the `split_to_right`` rule, so that the `table_id`` originally passed in will be split to left.
+            table_id = table_id.strict_add(1);
+            vnode = VirtualNode::ZERO;
+        }
+
+        FullKey::new(
+            TableId::from(table_id),
+            TableKey(vnode.to_be_bytes().to_vec()),
+            HummockEpoch::MIN,
```

Comment:
Is it HummockEpoch::MAX?

Reply:
Please update the PR description and relevant docs in the codes as well.